File: backend/internal/models/expense.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

// Expense ค่าใช้จ่าย
type Expense struct {
	ID          primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	ShopID      string             `bson:"shop_id" json:"shop_id"`
	TripID      string             `bson:"trip_id,omitempty" json:"trip_id"`
	VehicleID   string             `bson:"vehicle_id" json:"vehicle_id"`
	DriverID    string             `bson:"driver_id" json:"driver_id"`
	Type        string             `bson:"type" json:"type"` // "fuel", "toll", "parking", "repair", "fine", "other"
	Description string             `bson:"description" json:"description"`
	Amount      float64            `bson:"amount" json:"amount"`

	FuelDetail *FuelDetail `bson:"fuel_detail,omitempty" json:"fuel_detail"`

	ReceiptURL string `bson:"receipt_url,omitempty" json:"receipt_url"`

	BCAccountEntry *BCAccountEntry `bson:"bc_account_entry,omitempty" json:"bc_account_entry"`

	RecordedBy string    `bson:"recorded_by" json:"recorded_by"`
	RecordedAt time.Time `bson:"recorded_at" json:"recorded_at"`
	CreatedAt  time.Time `bson:"created_at" json:"created_at"`
}

// FuelDetail รายละเอียดเติมน้ำมัน
type FuelDetail struct {
	Liters        float64 `bson:"liters" json:"liters"`
	PricePerLiter float64 `bson:"price_per_liter" json:"price_per_liter"`
	OdometerKm    int     `bson:"odometer_km" json:"odometer_km"`
	Station       string  `bson:"station" json:"station"`
	FuelType      string  `bson:"fuel_type" json:"fuel_type"` // "ดีเซล B7", "แก๊สโซฮอล์ 95"
}

File: flutter/packages/fleet_core/lib/models/expense.dart
/// Expense model — ค่าใช้จ่าย
class Expense {
  final String id;
  final String shopId;
  final String? tripId;
  final String? vehicleId;
  final String? driverId;
  final String type; // "fuel", "toll", "parking", "repair", "fine", "other"
  final String? description;
  final double amount;
  final double? fuelLiters;
  final double? fuelPricePerLiter;
  final int? odometerKm;
  final String? receiptUrl;
  final DateTime? recordedAt;
  final DateTime createdAt;

  Expense({
    required this.id,
    required this.shopId,
    this.tripId,
    this.vehicleId,
    this.driverId,
    required this.type,
    this.description,
    required this.amount,
    this.fuelLiters,
    this.fuelPricePerLiter,
    this.odometerKm,
    this.receiptUrl,
    this.recordedAt,
    required this.createdAt,
  });

  factory Expense.fromJson(Map<String, dynamic> json) {
    return Expense(
      id: json['id'] as String,
      shopId: json['shop_id'] as String,
      tripId: json['trip_id'] as String?,
      vehicleId: json['vehicle_id'] as String?,
      driverId: json['driver_id'] as String?,
      type: json['type'] as String,
      description: json['description'] as String?,
      amount: (json['amount'] as num).toDouble(),
      fuelLiters: (json['fuel_liters'] as num?)?.toDouble(),
      fuelPricePerLiter: (json['fuel_price_per_liter'] as num?)?.toDouble(),
      odometerKm: json['odometer_km'] as int?,
      receiptUrl: json['receipt_url'] as String?,
      recordedAt: json['recorded_at'] != null ? DateTime.parse(json['recorded_at']) : null,
      createdAt: DateTime.parse(json['created_at']),
    );
  }

  Map<String, dynamic> toJson() => {
    'trip_id': tripId,
    'vehicle_id': vehicleId,
    'driver_id': driverId,
    'type': type,
    'description': description,
    'amount': amount,
    'fuel_liters': fuelLiters,
    'fuel_price_per_liter': fuelPricePerLiter,
    'odometer_km': odometerKm,
  };

  /// ประเภทค่าใช้จ่ายเป็นภาษาไทย
  String get typeThai {
    switch (type) {
      case 'fuel': return 'น้ำมัน';
      case 'toll': return 'ทางด่วน';
      case 'parking': return 'ที่จอดรถ';
      case 'repair': return 'ซ่อมบำรุง';
      case 'fine': return 'ค่าปรับ';
      case 'other': return 'อื่นๆ';
      default: return type;
    }
  }
}
